### PR TITLE
Update readme and create usage examples (closes #17)

### LINKED ISSUE #17: Update the readme (and create usage examples)
## Summary
Update the readme  file

## Motivation
The project Readme is describing itself as LightGBM. It is understood that this is a consequence of the fork, but  should be updated.

### PR DESCRIPTION
**Pull request by AndreFCruz**
Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,161 @@
...
 
-_Under construction; will be released in mid September._
+> **Under Construction**; release date: 16th of September.
+
+FairGBM is an easy-to-use and lightweight fairness-aware ML algorithm.
+Specifically, it enables **constrained optimization** with Gradient Boosting models.
+
+FairGBM builds upon the popular [LightGBM](https://github.com/microsoft/LightGBM) algorithm and adds customizable
+constraints for group-wise fairness (_e.g._, equal opportunity, predictive equality) and other global goals (_e.g._,
+specific Recall or FPR targets).
```
Comment (AlbertoEAF) [inline README.md]:
```suggestion
specific Recall or FPR prediction targets).
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,161 @@
...
+
+## Examples
+
+A short Python code snippet:
+```python
+from fairgbm import FairGBMClassifier
+
+# Instantiate
+fairgbm_clf = FairGBMClassifier(
+    "constraint_type"="FNR",    # constraint on equal group-wise TPR (equal opportunity)
```
Comment (AlbertoEAF) [inline README.md]: Why the quotes on the parameter names?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,161 @@
...
+You can use FairGBM to equalize the following metrics across protected groups:
+- Equalize FNR (equivalent to equalizing TPR or Recall)
+    - also known as _equal opportunity_ [(Hardt et al., 2016)](#external-references)
+- Equalize FPR (equivalent to equalizing TNR or Specificity)
+    - also known as _predictive equality_ [(Corbett-Davies et al., 2017)](#external-references)
+- Equalize both FNR and FPR simultaneously
+    - also known as _equal odds_ [(Hardt et al., 2016)](#external-references)
+
+<!--
+Take the following hypothetical example:
```
Comment (AlbertoEAF) [inline README.md:107]: Will we use this example?

File: examples/README.md [hunk 4]
```diff
@@ -1,62 +1,125 @@
...
+## Using Python
+
+**[Recommended]** Using the sklearn-style API:
+
+```python
+from fairgbm import FairGBMClassifier
+
+# Instantiate
+fairgbm_clf = FairGBMClassifier(
+    "constraint_type"="FNR",    # constraint on equal group-wise TPR (equal opportunity)
```
Comment (AlbertoEAF) [inline examples/README.md]: Same as above

**Review by AlbertoEAF — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (AndreFCruz) [reply README.md]: ups, this was copied from a dictionary :sweat_smile: 

thanks!

File: README.md (hunk 3 above)
Comment (AndreFCruz) [reply README.md:107]: That's what I'm trying to figure out. I wrote it and left it for PR reviewers, I think it is useful information but we should strive to keep the main README as short as possible; perhaps this should be somewhere else or we should just point people to the paper

**Pull request merged**